src/main/java/com/synerzip/projectmanagementapp/controller/ProjectController.java

```diff
@@ -23,17 +44,41 @@
 
 	@GET
 	@Produces(MediaType.APPLICATION_JSON)
-	@Path("{projectId}")
-	public Project getProject(@PathParam("projectId") long projectId) {
-		return service.getProject(projectId);
+	@Path("/{projectId}")
+	public Response getProject(@PathParam("projectId") long projectId) {
+		Project project = service.getProject(projectId);
+		if (project == null) {
+			throw new EntityNotFoundException("record not found with id " + projectId);
```

Comment:
why don't throwing this error from service and handling it in exception handler/mapper